### ISSUE TITLE
Login fails after update to new release

### ISSUE THREAD

**Issue by networxnet** [skodaconnect/homeassistant-myskoda#372]
After updating to newest release 1.12.1 login fails.

"Ein unerwarteter Fehler ist aufgetreten" ist Show after entering E-Mail and Password.

Update:

Seems you have to agree some new annoying marketing consents from skoda.

Log off from app, login again and answer the marketing consents. After this login in HA is possible again.

**Comment by pawelaugustyn** [comment]
It's worth noting that the _old_ Skoda Connect integration showed a more precise error (`Missing consent for client "technical" scopes`) while the new integration limited the output to `Config entry 'my-mail@mydomain.com' for myskoda integration could not authenticate`

**Comment by WebSpider** [comment]
It seems that we only fixed this error occurring in a running instance, not upon startup/restart.

**Comment by pawelaugustyn** [comment]
I'm doing a daily backup that includes both full stop and start of my HA instance (for the db integrity). That's why I noticed that

**Comment by henrik-farre** [comment]
Log:
```
2024-12-01 16:20:13.194 WARNING (MainThread) [homeassistant.config_entries] Config entry 'xxx@redacted.xxx' for myskoda integration could not authenticate
2024-12-01 16:20:13.946 ERROR (MainThread) [homeassistant] Error doing job: Unclosed client session (None)
2024-12-01 16:20:13.946 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/myskoda/auth/authorization.py", line 77, in authorize
    self.idk_session = await self._get_idk_session()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myskoda/auth/authorization.py", line 222, in _get_idk_session
    authentication = await self._enter_password(login_meta)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myskoda/auth/authorization.py", line 167, in _enter_password
    location = response.headers["Location"]
               ~~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'Location'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/config/custom_components/myskoda/__init__.py", line 60, in async_setup_entry
    await myskoda.connect(config.data["email"], config.data["password"])
  File "/usr/local/lib/python3.12/site-packages/myskoda/myskoda.py", line 128, in connect
    await self.authorization.authorize(email, password)
  File "/usr/local/lib/python3.12/site-packages/myskoda/auth/authorization.py", line 79, in authorize
    raise AuthorizationFailedError from ex
myskoda.auth.authorization.AuthorizationFailedError
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 635, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/myskoda/__init__.py", line 63, in async_setup_entry
    raise ConfigEntryAuthFailed from exc
homeassistant.exceptions.ConfigEntryAuthFailed
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1088, in _async_init_reauth
    result = await hass.config_entries.flow.async_init(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1301, in async_init
    flow, result = await self._async_init(flow_id, handler, context, data)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1336, in _async_init
    result = await self._async_handle_step(flow, flow.init_step, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 513, in _async_handle_step
    self._raise_if_step_does_not_exist(flow, step_id)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 592, in _raise_if_step_does_not_exist
    raise UnknownStep(
homeassistant.data_entry_flow.UnknownStep: Handler ConfigEntriesFlowManager doesn't support step reauth
```

**Comment by droosma** [comment]
I'm running into the same issue

**Comment by nils512** [comment]
is there any workaround?

**Comment by WebSpider** [comment]
Log off from app, login again and answer the marketing consents. After this login in HA is possible again.

**Comment by networxnet** [comment]
> is there any workaround?

Did you read my first post?

**Comment by nils512** [comment]
> > is there any workaround?
> 
> Did you read my first post?

Sorry, now i did. Works again :-)
Thank you!

**Comment by WebSpider** [comment]
> It's worth noting that the _old_ Skoda Connect integration showed a more precise error (`Missing consent for client "technical" scopes`) while the new integration limited the output to `Config entry 'my-mail@mydomain.com' for myskoda integration could not authenticate`

Yeah, the SkodaConnect integration was more mature in these aspects, absolutely. 
We're getting there ;-)

**Comment by dvx76** [comment]
To confirm, the bug to be fixed here is to also catch the TermsAndConditons exception in the ConfigFlow instead of just the init?

**Comment by WebSpider** [comment]
> To confirm, the bug to be fixed here is to also catch the TermsAndConditons exception in the ConfigFlow instead of just the init?

That, and we need to expand the throwing of the exception a bit as well. Last occurance was a bit different from previous ones.

**Comment by andsee** [comment]
I think this bug covers the exception I'm getting after downloading and trying configuration for the first time? If not let me know and I'll add a separate issue:

Logger: custom_components.myskoda.config_flow
Source: custom_components/myskoda/config_flow.py:108
integration: MySkoda (documentation, issues)
First occurred: 14:57:17 (2 occurrences)
Last logged: 15:34:41

Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/myskoda/config_flow.py", line 108, in async_step_user
    await validate_input(self.hass, user_input)
  File "/config/custom_components/myskoda/config_flow.py", line 65, in validate_input
    await hub.connect(data["email"], data["password"])
  File "/usr/local/lib/python3.13/site-packages/myskoda/myskoda.py", line 128, in connect
    await self.authorization.authorize(email, password)
  File "/usr/local/lib/python3.13/site-packages/myskoda/auth/authorization.py", line 77, in authorize
    self.idk_session = await self._get_idk_session()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/myskoda/auth/authorization.py", line 222, in _get_idk_session
    authentication = await self._enter_password(login_meta)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/myskoda/auth/authorization.py", line 165, in _enter_password
    raise TermsAndConditionsError(location)
myskoda.auth.authorization.TermsAndConditionsError: /signin-service/v1/fffffffffffffffffffffffffffff@apps_vw-dilab_com/terms-and-conditions?relayState=ffffffffffffffffffffffffffffffffff&canChangeCountryOfResidence=true&userId=ffffffffffffffffffffffffffffffffffffff

**Comment by dvx76** [comment]
> > To confirm, the bug to be fixed here is to also catch the TermsAndConditons exception in the ConfigFlow instead of just the init?
> 
> That, and we need to expand the throwing of the exception a bit as well. Last occurance was a bit different from previous ones.

And perhaps also adding a generic note in the README stating _if login issues -> log out and back in with the App_

**Comment by WebSpider** [comment]
Fixed in v1.16 and later
